proposals/lip-0052.md

```diff
@@ -30,329 +31,345 @@ Therefore, providing a unified module to handle, transfer and modify NFTs is a n
+                        "pattern": "^[a-zA-Z0-9]*$",
+                        "fieldNumber": 1
+                    },
+                    "attributes": {
+                        "dataType": "bytes",
+                        "fieldNumber": 2
+                    }
+                }
+            }
         }
     }
 }
 ```
 
-#### User Substore
+Here, the `attributesArray` array is lexicographically ordered by `module`, which guarantees that serialization is consistent across nodes maintaining the chain.
 
-* The store prefix is set to `STORE_PREFIX_USER`.
-* Each store key is a 20-byte address and an NFT ID: `address||uint32be(chainID)||uint32be(collection)||uint64be(index)`.
-* Each store value is the serialization of an object following `userStoreSchema`.
+If, for some `module`, a state transition deletes the `attributes` property , the corresponding entry in `attributesArray` is removed.
 
-```java
-userStoreSchema = {
-    "type": "object",
-    "required": ["exists"],
-    "properties": {
-        "exists": {
-            "dataType": "boolean",
-            "fieldNumber": 1
-        }
-    }
-}
-```
+If, for some `module`, a state transition creates the `attributes` property of a non-existent store entry, this entry is created following `NFTStoreSchema` with the `attributesArray` entry set accordingly.
 
-#### Collection Substore
+#### User Substore
 
-* The store prefix is set to `STORE_PREFIX_COLLECTION`.
-* Each store key is a collection: `uint32be(collection)`.
-* Each store value is the serialization of an object following `collectionStoreSchema`.
+- The substore prefix is set to `SUBSTORE_PREFIX_USER`.
+- Each store key is a concatenation of an address and a NFT ID: `address + nftID`.
+- Each store value is the serialization of an object following `userStoreSchema`.
 
 ```java
-collectionStoreSchema = {
+userStoreSchema = {
     "type": "object",
-    "required": ["nextAvailableIndex"],
+    "required": ["lockingModule"],
     "properties": {
-        "nextAvailableIndex": {
-            "dataType": "uint64",
+        "lockingModule": {
+            "dataType": "string",
+            "minLength": MIN_MODULE_NAME_LENGTH,
+            "maxLength": MAX_MODULE_NAME_LENGTH,
+            "pattern": "^[a-zA-Z0-9]*$]",
             "fieldNumber": 1
-        }
+        },
     }
 }
 ```
 
-#### Available Collection Substore
+The default value for the `lockingModule` property is `NFT_NOT_LOCKED`.
 
-* The store prefix is set to `STORE_PREFIX_AVAILABLE_COLLECTION`.
-* Each store key is the empty bytes.
-* Each store value is the serialization of an object following `availableCollectionStoreSchema`.
+#### Escrow Substore
+
+- The substore prefix is set to `SUBSTORE_PREFIX_ESCROW`.
+- Each store key is the identifier of the chain to which the NFTs are escrowed, and the NFT ID of the escrowed NFT: `escrowedChainID + nftID`.
+- Each store value is the serialization of an object following `escrowStoreSchema`.
```

Comment:
Here I think we should use the empty schema for the store value (basically we store nothing). We indicate that the NFT is escrowed simply by the existence of the entry in the substore, which we should delete once the NFT is unescrowed.

Reply:
I agree and have changed escrowing accordingly.